## Cleaning on the Datasets will be performed in this Notebook

In [155]:
# Imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Load Data

In [156]:
# 2022-2023
hms_23 = pd.read_csv("../data/raw/2022-2023/HMS_2022-2023_PUBLIC_instchars.csv")


C:\Users\noahh\AppData\Local\Temp\ipykernel_13300\2745917235.py:2: DtypeWarning: Columns (11: gender_text, 37: black_other_text, 43: asian_other_text, 50: his_other_text, 56: st_tps, 63: st_otherdoc_text, 78: degree_other_text, 101: covid_clsformat_text, 105: enroll_text, 135: residenc_text, 186: sib_other_text, 203: alc_binge_othgen, 240: dx_bip_4_text, 248: dx_ax_6_text, 251: dx_ocd_3, 255: dx_ocd_6_text, 260: dx_trauma_4_text, 266: dx_neurodev_4_text, 270: dx_ea_4, 273: dx_ea_6_text, 276: dx_psy_3, 277: dx_psy_4, 278: dx_psy_5, 279: dx_psy_6, 282: dx_perso_1, 283: dx_perso_2, 287: dx_perso_6, 290: dx_perso_9, 291: dx_perso_10, 292: dx_perso_11, 293: dx_perso_12, 298: dx_sa_3, 299: dx_sa_2_text, 310: disab_1_9_text, 322: talk1_8_text, 412: stim_reason_text, 418: antidep_reason_text, 422: antipsy_reason_4, 423: antipsy_reason_5, 430: antianx_reason_text, 432: mood_reason_2, 434: mood_reason_4, 436: mood_reason_text, 440: slpmed_reason_4, 442: slpmed_reason_text, 448: othmed_reason_tex

In [157]:
# 2023-2024
hms_24 = pd.read_csv("../data/raw/2023-2024/HMS_2023-2024_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_13300\2315900744.py:2: DtypeWarning: Columns (12: gender_text, 23: sexual_text, 38: black_other_text, 44: asian_other_text, 51: his_other_text, 64: st_otherdoc_text, 109: covid_clsformat_text, 113: enroll_text, 161: field_other_tech_text, 170: living_sit_text, 234: sib_other_text, 300: internet_3_8_TEXT, 348: dx_bip_4_text, 356: dx_ax_6_text, 364: dx_ocd_6_text, 370: dx_trauma_4_text, 376: dx_neurodev_4_text, 384: dx_ea_6_text, 392: dx_psy_6_text, 405: dx_perso_11_text, 410: dx_sa_2_text, 523: stim_reason_TEXT, 529: antianx_reason_TEXT, 536: meds_w_3_text, 540: stim_prescriber_7_TEXT, 571: why_tx_8_text, 608: inf_7_text, 641: ins_ina_8_text, 688: infoseek_oth_text, 699: witness_6_text, 712: inter_y_6_text, 721: inter_n_6_text, 731: inter_n_why_9_text, 790: drug_other_text, 872: ab_emo_othcont_text, 940: hostcli_other_text, 970: excl_pers_other_text, 999: campus_oth_text, 1064: sa_occur_loc6_text, 1081: sa_loc_oth_text, 1094: sa_who_oth_text, 

In [158]:
# 2024-2025
hms_25 = pd.read_csv("../data/raw/2024-2025/HMS_2024-2025_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_13300\1171019825.py:2: DtypeWarning: Columns (13: gender_text, 25: sexual_text, 53: state_9thgrade, 54: state_10thgrade, 55: state_11thgrade, 56: state_12thgrade, 57: year_9thgrade, 58: year_10thgrade, 59: year_11thgrade, 60: year_12thgrade, 881: Grad, 1475: Peer, 1590: StartDate2) have mixed types. Specify dtype option on import or set low_memory=False.
  hms_25 = pd.read_csv("../data/raw/2024-2025/HMS_2024-2025_PUBLIC_instchars.csv")


In [159]:
# 2025-2026
hms_26 = pd.read_csv("../data/raw/2025-2026/HMS_2025-2026_PUBLIC_instchars.csv")

C:\Users\noahh\AppData\Local\Temp\ipykernel_13300\4153514391.py:2: DtypeWarning: Columns (12: gender_text, 24: sexual_text, 55: race_other_text, 61: black_other_text, 67: asian_other_text, 74: his_other_text, 81: pi_other_text, 85: residenc_text, 87: living_sit_text, 115: clsformat_text, 131: degree_other_tech, 135: enroll_text, 157: field_other_text, 184: field_other_tech_text, 255: sib_other_text, 318: recovery_path_text, 328: recovery_service_text, 337: recovery_aware_text, 359: dx_dep_4_text, 365: dx_bip_4_text, 373: dx_ax_6_text, 381: dx_ocd_6_text, 387: dx_trauma_4_text, 393: dx_neurodev_4_text, 401: dx_ea_6_text, 409: dx_psy_6_text, 422: dx_perso_11_text, 427: dx_sa_2_text, 444: ADHD2_4_TEXT, 446: ADHD3_text, 463: A988_followup_text, 465: A988_reason_text, 478: talk1_8_text, 502: ADHD_therapy_loc_text, 575: stim_reason_TEXT, 581: antianx_reason_TEXT, 588: meds_w_3_text, 601: ADHD_meds_text, 622: why_tx_6_text, 623: why_tx_8_text, 624: why_tx_12_text, 663: inf_7_text, 696: ins_in

In [160]:
# selected variables

selected_vars = [
    # Outcomes
    "flourish",
    "deprawsc",
    "anx_score",

    # Main factors
    "fincur",
    "finpast",
    "belong1",
    "discrim_race",
    "discrim_culture",
    "discrim_gender",
    "discrim_sexual",
    "alc_any",

    # Controls
    "age",
    "international",
    "educ_par1",
    "educ_par2",
    "enroll",

    # Survey weight
    "nrweight"]

In [161]:
# create copies of data
hms_23_selected = hms_23[selected_vars].copy()
hms_24_selected = hms_24[selected_vars].copy()
hms_25_selected = hms_25[selected_vars].copy()
hms_26_selected = hms_26[selected_vars].copy()


In [162]:
# create year columns
hms_23_selected["survey_year"] = "2022-2023"
hms_24_selected["survey_year"] = "2023-2024"
hms_25_selected["survey_year"] = "2024-2025"
hms_26_selected["survey_year"] = "2025-2026"


In [163]:
# Join the tables together

hms_all = pd.concat([hms_23_selected, hms_24_selected, hms_25_selected, hms_26_selected],
                    ignore_index = True)
hms_all.head()

,flourish,deprawsc,anx_score,fincur,finpast,belong1,discrim_race,discrim_culture,discrim_gender,discrim_sexual,alc_any,age,international,educ_par1,educ_par2,enroll,nrweight,survey_year
0,45.0,9.0,13.0,2.0,3.0,3.0,0.0,0.0,0.0,0.0,1.0,21.0,0.0,5.0,3.0,1.0,3.662791,2022-2023
1,NaN,NaN,NaN,3.0,3.0,NaN,NaN,NaN,NaN,NaN,NaN,21.0,0.0,4.0,3.0,1.0,5.362319,2022-2023
2,51.0,2.0,2.0,4.0,5.0,6.0,0.0,0.0,0.0,0.0,1.0,41.0,0.0,6.0,5.0,2.0,3.662791,2022-2023
3,50.0,6.0,3.0,3.0,4.0,3.0,0.0,0.0,0.0,0.0,1.0,24.0,0.0,4.0,3.0,1.0,5.362319,2022-2023
4,55.0,3.0,NaN,2.0,1.0,5.0,0.0,0.0,0.0,0.0,0.0,27.0,0.0,3.0,5.0,1.0,3.662791,2022-2023


## Inspect Joined Data

In [164]:
# shape
hms_all.shape

(333347, 18)

In [165]:
# check how many per year
hms_all["survey_year"].value_counts()

survey_year
2023-2024    104729
2024-2025     84735
2022-2023     76406
2025-2026     67477
Name: count, dtype: int64

In [166]:
# info
hms_all.info()

<class 'pandas.DataFrame'>
RangeIndex: 333347 entries, 0 to 333346
Data columns (total 18 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   flourish         302632 non-null  float64
 1   deprawsc         295689 non-null  float64
 2   anx_score        294916 non-null  float64
 3   fincur           322082 non-null  float64
 4   finpast          321710 non-null  float64
 5   belong1          314931 non-null  float64
 6   discrim_race     310647 non-null  float64
 7   discrim_culture  310250 non-null  float64
 8   discrim_gender   309727 non-null  float64
 9   discrim_sexual   309578 non-null  float64
 10  alc_any          296716 non-null  float64
 11  age              333347 non-null  float64
 12  international    328803 non-null  float64
 13  educ_par1        320653 non-null  float64
 14  educ_par2        315405 non-null  float64
 15  enroll           316938 non-null  float64
 16  nrweight         333347 non-null  float64
 17  su

In [167]:
# types
hms_all.dtypes
# will be changing some data types to integer

flourish           float64
deprawsc           float64
anx_score          float64
fincur             float64
finpast            float64
belong1            float64
discrim_race       float64
discrim_culture    float64
discrim_gender     float64
discrim_sexual     float64
alc_any            float64
age                float64
international      float64
educ_par1          float64
educ_par2          float64
enroll             float64
nrweight           float64
survey_year            str
dtype: object

## Valid Data

In [168]:
# describe
hms_all.describe()

,flourish,deprawsc,anx_score,fincur,finpast,belong1,discrim_race,discrim_culture,discrim_gender,discrim_sexual,alc_any,age,international,educ_par1,educ_par2,enroll,nrweight
count,302632.000000,295689.000000,294916.000000,322082.000000,321710.000000,314931.000000,310647.000000,310250.000000,309727.000000,309578.000000,296716.000000,333347.000000,328803.000000,320653.000000,315405.000000,316938.000000,333347.000000
mean,43.593182,8.496654,7.689793,2.746406,3.143632,2.920198,0.061481,0.054775,0.090383,0.037038,0.453420,23.590323,0.076182,5.062710,4.941551,1.142413,1.734507
std,8.943413,6.269181,5.826097,1.118866,1.253971,1.354833,0.240212,0.227541,0.286730,0.188854,0.497826,7.677082,0.265290,1.818527,1.871557,0.376749,2.592178
min,8.000000,0.000000,0.000000,1.000000,1.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,18.000000,0.000000,1.000000,1.000000,1.000000,0.142613
25%,39.000000,4.000000,3.000000,2.000000,2.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,19.000000,0.000000,4.000000,3.000000,1.000000,0.631376
50%,46.000000,7.000000,7.000000,3.000000,3.000000,3.000000,0.000000,0.000000,0.000000,0.000000,0.000000,21.000000,0.000000,6.000000,6.000000,1.000000,1.077505
75%,50.000000,12.000000,12.000000,3.000000,4.000000,4.000000,0.000000,0.000000,0.000000,0.000000,1.000000,25.000000,0.000000,7.000000,6.000000,1.000000,1.970260
max,56.000000,27.000000,21.000000,5.000000,5.000000,6.000000,1.000000,1.000000,1.000000,1.000000,1.000000,120.000000,1.000000,8.000000,8.000000,3.000000,277.500000


In [169]:
# are outcomes in range
hms_all[["flourish", "deprawsc", "anx_score"]].describe().loc[["min", "max"]]

,flourish,deprawsc,anx_score
min,8.0,0.0,0.0
max,56.0,27.0,21.0


In [170]:
# are factors in range
hms_all[["fincur", "finpast", "belong1", "discrim_race",
         "discrim_culture", "discrim_gender", "discrim_sexual", "alc_any"]].describe().loc[["min", "max"]]

,fincur,finpast,belong1,discrim_race,discrim_culture,discrim_gender,discrim_sexual,alc_any
min,1.0,1.0,1.0,0.0,0.0,0.0,0.0,0.0
max,5.0,5.0,6.0,1.0,1.0,1.0,1.0,1.0


In [171]:
# are outcome numbers valid
outcome_cols = ["flourish", "deprawsc", "anx_score"]

for col in outcome_cols:
    print(col, sorted(hms_all[col].dropna().unique()))

flourish [np.float64(8.0), np.float64(9.0), np.float64(10.0), np.float64(11.0), np.float64(12.0), np.float64(13.0), np.float64(14.0), np.float64(15.0), np.float64(16.0), np.float64(17.0), np.float64(18.0), np.float64(19.0), np.float64(20.0), np.float64(21.0), np.float64(22.0), np.float64(23.0), np.float64(24.0), np.float64(25.0), np.float64(26.0), np.float64(27.0), np.float64(28.0), np.float64(29.0), np.float64(30.0), np.float64(31.0), np.float64(32.0), np.float64(33.0), np.float64(34.0), np.float64(35.0), np.float64(36.0), np.float64(37.0), np.float64(38.0), np.float64(39.0), np.float64(40.0), np.float64(41.0), np.float64(42.0), np.float64(43.0), np.float64(44.0), np.float64(45.0), np.float64(46.0), np.float64(47.0), np.float64(48.0), np.float64(49.0), np.float64(50.0), np.float64(51.0), np.float64(52.0), np.float64(53.0), np.float64(54.0), np.float64(55.0), np.float64(56.0)]
deprawsc [np.float64(0.0), np.float64(1.0), np.float64(2.0), np.float64(3.0), np.float64(4.0), np.float64(5.0)

In [172]:
# are factor numbers valid
factor_cols = ["fincur", "finpast", "belong1", "discrim_race",
               "discrim_culture", "discrim_gender", "discrim_sexual","alc_any"]

for col in factor_cols:
    print(col, sorted(hms_all[col].dropna().unique()))

fincur [np.float64(1.0), np.float64(2.0), np.float64(3.0), np.float64(4.0), np.float64(5.0)]
finpast [np.float64(1.0), np.float64(2.0), np.float64(3.0), np.float64(4.0), np.float64(5.0)]
belong1 [np.float64(1.0), np.float64(2.0), np.float64(3.0), np.float64(4.0), np.float64(5.0), np.float64(6.0)]
discrim_race [np.float64(0.0), np.float64(1.0)]
discrim_culture [np.float64(0.0), np.float64(1.0)]
discrim_gender [np.float64(0.0), np.float64(1.0)]
discrim_sexual [np.float64(0.0), np.float64(1.0)]
alc_any [np.float64(0.0), np.float64(1.0)]


In [173]:
# are control number valid
control_cols = [
    "age",
    "international",
    "educ_par1",
    "educ_par2",
    "enroll"
]

for col in control_cols:
    print(f"\n{col}")
    print("Min:", hms_all[col].min())
    print("Max:", hms_all[col].max())
    print("Unique:", sorted(hms_all[col].dropna().unique()))


age
Min: 18.0
Max: 120.0
Unique: [np.float64(18.0), np.float64(18.6), np.float64(19.0), np.float64(19.5), np.float64(20.0), np.float64(21.0), np.float64(21.1), np.float64(22.0), np.float64(23.0), np.float64(23.312), np.float64(24.0), np.float64(25.0), np.float64(26.0), np.float64(26.9), np.float64(27.0), np.float64(28.0), np.float64(29.0), np.float64(30.0), np.float64(31.0), np.float64(32.0), np.float64(33.0), np.float64(34.0), np.float64(35.0), np.float64(36.0), np.float64(37.0), np.float64(38.0), np.float64(39.0), np.float64(40.0), np.float64(41.0), np.float64(42.0), np.float64(43.0), np.float64(44.0), np.float64(45.0), np.float64(46.0), np.float64(47.0), np.float64(48.0), np.float64(49.0), np.float64(50.0), np.float64(51.0), np.float64(52.0), np.float64(53.0), np.float64(54.0), np.float64(55.0), np.float64(56.0), np.float64(57.0), np.float64(58.0), np.float64(59.0), np.float64(60.0), np.float64(61.0), np.float64(62.0), np.float64(63.0), np.float64(64.0), np.float64(65.0), np.float6

In [174]:
# Ages that contain a decimal
decimal_ages = hms_all[
    hms_all["age"] % 1 != 0
]

decimal_ages["age"].value_counts().sort_index()

age
18.600    1
19.500    1
21.100    1
23.312    1
26.900    1
Name: count, dtype: int64

In [175]:
# ages >= 80
hms_all[hms_all["age"] >= 80]["age"].value_counts().sort_index()

age
80.0     13
81.0      7
82.0      6
83.0      5
84.0      7
85.0      5
86.0      2
87.0      2
88.0      1
89.0      1
90.0      3
95.0      1
96.0      1
97.0      1
98.0      1
99.0     12
100.0     3
104.0     1
111.0     1
118.0     1
119.0     1
120.0     3
Name: count, dtype: int64

All numbers for both outcomes and factors are valid in the specified ranges wihin the data dictionary. 

In [176]:
# what years have decimal ages
decimal_ages.groupby(
    ["survey_year", "age"]
).size()

survey_year  age   
2022-2023    19.500    1
2023-2024    18.600    1
             21.100    1
             23.312    1
             26.900    1
dtype: int64

In [177]:
# What years have older respondents
hms_all[hms_all["age"] >= 80].groupby(
    ["survey_year", "age"]
).size()

survey_year  age  
2022-2023    80.0     2
             96.0     1
             97.0     1
             99.0     3
             100.0    1
             104.0    1
             120.0    1
2023-2024    80.0     1
             81.0     2
             83.0     1
             84.0     2
             90.0     1
             100.0    2
             111.0    1
             118.0    1
             119.0    1
             120.0    2
2024-2025    80.0     9
             81.0     4
             82.0     3
             83.0     4
             84.0     3
             85.0     5
             86.0     2
             87.0     2
             88.0     1
             89.0     1
             90.0     1
             95.0     1
             98.0     1
             99.0     4
2025-2026    80.0     1
             81.0     1
             82.0     3
             84.0     2
             90.0     1
             99.0     5
dtype: int64

In [178]:
# how many have age greater than 100
hms_all[hms_all["age"] >= 100].shape[0]
# percent 
hms_all[hms_all["age"] >= 100].shape[0] / len(hms_all) * 100

0.002999877005042793

`age` is ranged from 18 to 120 years. Five of the respondents have decimal ages and 10 (0.003% of total combined data) have reported ages of 100 or older. The question asked for `age` is *"How old are you?"*, which does not specify a range of an integer number so it still follows the data dicionary. HMS codebook only spcifies a minimum age requirement not an upper limit or an integer response; age values will be kept without further modification.

## Changing Data Types

In [179]:
# types (again)
hms_all.dtypes

flourish           float64
deprawsc           float64
anx_score          float64
fincur             float64
finpast            float64
belong1            float64
discrim_race       float64
discrim_culture    float64
discrim_gender     float64
discrim_sexual     float64
alc_any            float64
age                float64
international      float64
educ_par1          float64
educ_par2          float64
enroll             float64
nrweight           float64
survey_year            str
dtype: object

In [180]:
# changing data types to int nullable except for age, nrweight, and survery year
int_cols = ["flourish", "deprawsc", "anx_score", "fincur",
            "finpast", "belong1", "discrim_race", "discrim_culture",
            "discrim_gender", "discrim_sexual", "alc_any", "international",
            "educ_par1", "educ_par2", "enroll"]

hms_all[int_cols] = hms_all[int_cols].astype("Int64")

In [181]:
# check types
hms_all.dtypes

flourish             Int64
deprawsc             Int64
anx_score            Int64
fincur               Int64
finpast              Int64
belong1              Int64
discrim_race         Int64
discrim_culture      Int64
discrim_gender       Int64
discrim_sexual       Int64
alc_any              Int64
age                float64
international        Int64
educ_par1            Int64
educ_par2            Int64
enroll               Int64
nrweight           float64
survey_year            str
dtype: object

## Missingness

In [182]:
# NA values
hms_all.isna().sum()

flourish           30715
deprawsc           37658
anx_score          38431
fincur             11265
finpast            11637
belong1            18416
discrim_race       22700
discrim_culture    23097
discrim_gender     23620
discrim_sexual     23769
alc_any            36631
age                    0
international       4544
educ_par1          12694
educ_par2          17942
enroll             16409
nrweight               0
survey_year            0
dtype: int64

In [183]:
# percentage of missingness per column
missing_summary = pd.DataFrame({
    "missing_count": hms_all.isna().sum(),
    "missing_percent": hms_all.isna().mean() * 100
})

missing_summary.sort_values("missing_percent", ascending=False)

,missing_count,missing_percent
anx_score,38431,11.528827
deprawsc,37658,11.296937
alc_any,36631,10.988849
flourish,30715,9.214122
discrim_sexual,23769,7.130408
discrim_gender,23620,7.085709
discrim_culture,23097,6.928816
discrim_race,22700,6.809721
belong1,18416,5.524573
educ_par2,17942,5.382379


In [184]:
# Missingness grouped by year as percent
hms_all.isna().groupby(hms_all["survey_year"]).mean() * 100

,flourish,deprawsc,anx_score,fincur,finpast,belong1,discrim_race,discrim_culture,discrim_gender,discrim_sexual,alc_any,age,international,educ_par1,educ_par2,enroll,nrweight,survey_year
survey_year,,,,,,,,,,,,,,,,,,
2022-2023,8.079208,10.173285,10.432427,1.895139,1.985446,4.376620,5.701123,5.826768,6.015234,6.038793,9.939010,0.0,0.858571,2.765490,4.010156,3.730074,0.0,0.0
2023-2024,7.578608,9.301149,9.406182,1.849535,1.987033,4.081009,5.101739,5.222049,5.360502,5.402515,8.849507,0.0,0.616830,2.364197,4.383695,3.368694,0.0,0.0
2024-2025,10.757066,13.172833,13.627191,6.359828,6.469582,7.187113,8.633977,8.767333,8.962058,9.025786,12.330206,0.0,1.782026,5.582109,7.043134,6.573435,0.0,0.0
2025-2026,11.100079,13.311202,13.429761,3.691628,3.789439,6.977192,8.425093,8.516976,8.619233,8.668139,13.813596,0.0,2.566801,5.001704,6.400699,6.611142,0.0,0.0


In [185]:
# Missingness percent for outcomes
hms_all[["flourish", "deprawsc", "anx_score"]].isna().mean() * 100

flourish      9.214122
deprawsc     11.296937
anx_score    11.528827
dtype: float64

In [186]:
# Missingness percent for factors
hms_all[["fincur", "finpast", "belong1", "discrim_race", "discrim_culture", "discrim_gender", "discrim_sexual", "alc_any"]].isna().mean() * 100

fincur              3.379361
finpast             3.490957
belong1             5.524573
discrim_race        6.809721
discrim_culture     6.928816
discrim_gender      7.085709
discrim_sexual      7.130408
alc_any            10.988849
dtype: float64

### Remove Rows where all Outcomes are Missing

In [220]:
# remove rows where flourish, deprawsc, and anx_score are missing
hms_all_na_removed = hms_all.dropna(subset = ["flourish", "deprawsc", "anx_score"], how = "all").copy()
hms_all_na_removed.shape

(306873, 18)

In [222]:
# see new data NA sum
hms_all_na_removed.isna().sum()

flourish            4241
deprawsc           11184
anx_score          11957
fincur              2796
finpast             3086
belong1              256
discrim_race        2273
discrim_culture     2644
discrim_gender      3154
discrim_sexual      3272
alc_any            10797
age                    0
international        409
educ_par1            850
educ_par2           5667
enroll               973
nrweight               0
survey_year            0
dtype: int64

In [209]:
# compare against old dataset
print(f"Removed: {hms_all.shape[0] - hms_all_na_removed.shape[0]} outcomes")
print(f"Removed: {round(((hms_all.shape[0] - hms_all_na_removed.shape[0]) / hms_all.shape[0] * 100), 3)}% of outcomes")


Removed: 26474 outcomes
Removed: 7.942% of outcomes


In [217]:
# check new dataset with outcome NAs removed when all three were missing
hms_all_na_removed.describe()

,flourish,deprawsc,anx_score,fincur,finpast,belong1,discrim_race,discrim_culture,discrim_gender,discrim_sexual,alc_any,age,international,educ_par1,educ_par2,enroll,nrweight
count,302632.0,295689.0,294916.0,304077.0,303787.0,306617.0,304600.0,304229.0,303719.0,303601.0,296076.0,306873.000000,306464.0,306023.0,301206.0,305900.0,306873.000000
mean,43.593182,8.496654,7.689793,2.743065,3.140371,2.920291,0.06106,0.054377,0.090199,0.036844,0.453779,23.682505,0.075285,5.063632,4.940861,1.141886,1.731492
std,8.943413,6.269181,5.826097,1.118562,1.254969,1.354383,0.239441,0.22676,0.286466,0.18838,0.49786,7.706359,0.26385,1.815204,1.868966,0.375704,2.600673
min,8.0,0.0,0.0,1.0,1.0,1.0,0.0,0.0,0.0,0.0,0.0,18.000000,0.0,1.0,1.0,1.0,0.142613
25%,39.0,4.0,3.0,2.0,2.0,2.0,0.0,0.0,0.0,0.0,0.0,19.000000,0.0,4.0,3.0,1.0,0.640625
50%,46.0,7.0,7.0,3.0,3.0,3.0,0.0,0.0,0.0,0.0,0.0,21.000000,0.0,6.0,6.0,1.0,1.077505
75%,50.0,12.0,12.0,3.0,4.0,4.0,0.0,0.0,0.0,0.0,1.0,25.000000,0.0,7.0,6.0,1.0,1.968354
max,56.0,27.0,21.0,5.0,5.0,6.0,1.0,1.0,1.0,1.0,1.0,120.000000,1.0,8.0,8.0,3.0,277.500000


In [216]:
# Missingness percent for outcomes
hms_all_na_removed[["flourish", "deprawsc", "anx_score"]].isna().mean() * 100

flourish     1.382005
deprawsc     3.644504
anx_score    3.896400
dtype: float64

In [215]:
# Missingness percent for factors
hms_all_na_removed[["fincur", "finpast", "belong1", "discrim_race", "discrim_culture", "discrim_gender", "discrim_sexual", "alc_any"]].isna().mean() * 100

fincur             0.911126
finpast            1.005628
belong1            0.083422
discrim_race       0.740697
discrim_culture    0.861594
discrim_gender     1.027787
discrim_sexual     1.066239
alc_any            3.518394
dtype: float64

In [218]:
# Missingness grouped by year as percent
hms_all_na_removed.isna().groupby(hms_all_na_removed["survey_year"]).mean() * 100

,flourish,deprawsc,anx_score,fincur,finpast,belong1,discrim_race,discrim_culture,discrim_gender,discrim_sexual,alc_any,age,international,educ_par1,educ_par2,enroll,nrweight,survey_year
survey_year,,,,,,,,,,,,,,,,,,
2022-2023,1.489586,3.733782,4.011502,0.093976,0.168315,0.056105,0.640999,0.768637,0.965005,0.980433,3.684690,0.0,0.050494,0.239849,1.482572,0.210393,0.0,0.0
2023-2024,1.603147,3.437058,3.548882,0.704490,0.826480,0.062011,0.573351,0.690258,0.833596,0.873243,3.158515,0.0,0.048796,0.261261,2.240543,0.196200,0.0,0.0
2024-2025,1.163247,3.838714,4.341916,2.114756,2.207555,0.107176,0.682264,0.823422,1.039080,1.095282,3.131617,0.0,0.079728,0.348974,1.825905,0.317606,0.0,0.0
2025-2026,1.173001,3.631032,3.762830,0.688644,0.764428,0.120266,1.202656,1.296562,1.402000,1.443187,4.393812,0.0,0.434933,0.255358,1.662301,0.637572,0.0,0.0


In [219]:
missing_summary_na_removed = pd.DataFrame({
    "missing_count": hms_all.isna().sum(),
    "missing_percent": hms_all.isna().mean() * 100
})

missing_summary_na_removed.sort_values("missing_percent", ascending=False)

,missing_count,missing_percent
anx_score,38431,11.528827
deprawsc,37658,11.296937
alc_any,36631,10.988849
flourish,30715,9.214122
discrim_sexual,23769,7.130408
discrim_gender,23620,7.085709
discrim_culture,23097,6.928816
discrim_race,22700,6.809721
belong1,18416,5.524573
educ_par2,17942,5.382379


...